docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md

```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
+
+### 4. Data Negotiation and Transfer Phase
+
+Once we have the submodel we are going to call the [`subprotocolBody`](#L233) url of the `endpoint interface` with name `SUBMODEL-3.0`. This will provide for us the asset id to negotiate with the EDC Provider. Once this asset is negotiated we will request for the `transfer` and `EDR` token will be sent to the backend by the EDC Provider, allowing us to query the dataplane url contained in the `href` field of the endpoint interface. And in this way we will retrieve the data using the `EDC Provider Data Plane Proxy`.
+
+
+## Technical Detailed Phases
+
+## 1. Discovery Phase + 2. Digital Twin Registry Search Phase
+
+At the discovery phase the digital product pass application (Eco Pass KIT) will search in the Discovery Service for the following keys:
+
+| Key                | Service Name          |
+| ------------------ | --------------------- |
+| bpn                | EDC Discovery Service |
+| manufacturerPartId | BPN Discovery Service |
+
+By searching for this keys the application will find the necessary services and will be able to search for the information needed to start the search for digital twin registries.
+
+After the discovery phase, the search for digital twin registries is one of the core phases to be done when retrieving data in Catena-X. Is necessary for filtering the EDCs which contain the digital twin registry assets, allowing an optimized search without high waiting times.
+
+Once the negotiation for the digital twin registries assets are done we would be able to retrieve a catalog for the user to search the serialized Id (key: partInstanceId, example value: batteryDMC_code).
+
+### Prerequisites
+
+The following information is required to enable the decentralized search for digital twin registries:
+
+| Name            | Example              | Description                                                                                                                                                                                                                                                                                       |
+| --------------- | -------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Search Id Type  | *manufacturerPartId* | The search id type is required first of all to know in which `BPN Discovery` services to search. It will be introduced in the `Discovery Service` and we will obtain a list of `BPN Discovery Endpoints`. After this same id will be introduce as the *`type`* attribute in each `BPN Discovery`. |
```

Comment:
"After this same id will be introduce**d** as the type [...]"

Reply:
done